web_services/foxx/brightid/db.js

```diff
@@ -581,17 +591,21 @@ function linkContextId(id, context, contextId, timestamp) {
   // remove testblocks if exists
   removeTestblock(contextId, 'link');
 
+  let user = getUserByContextId(coll, contextId);
+  if (user && user != id) {
+    throw 'contextId is duplicate';
+  }
```

Comment:
If this throw is allowed to bubble up and return to the user, the Error needs to have an ErrorNum.  In fact, we should define an Error object, class or interface that is suitable for returning to the API, which has an ErrorNum in the constructor and enforces (in the constructor) that it's in a list of defined numbers. These defined numbers could be an enum for readability.  See [Custom Error Types](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#Custom_Error_Types).

If we want to sometimes catch an Error instead of returning it in the API, we should extend the above Error class or the base Error class and just catch the exception we're looking for instead of catching all Errors and doing a comparison.

See section 5. in [this article](http://javierferrer.me/exceptions-vs-error-codes/) and [this issue in the BrightID repo](https://github.com/BrightID/BrightID/issues/702) for more information.

Reply:
I don't expect all the error handling to be fixed in this release, but I wanted to point it out.  I added another issue for it here https://github.com/BrightID/BrightID-Node/issues/177

Reply:
We will resolve issue #177 in the next release